core/src/main/scala/com/microsoft/azure/synapse/ml/codegen/Wrappable.scala

```diff
@@ -33,7 +33,7 @@ trait BaseWrappable extends Params {
       .find(c => Set("Estimator", "ProbabilisticClassifier", "Predictor", "BaseRegressor", "Ranker")(
         c.getSuperclass.getSimpleName))
       .get
-    val typeArgs = thisStage.getClass.getGenericSuperclass.asInstanceOf[ParameterizedType].getActualTypeArguments
+    val typeArgs = superClass.getGenericSuperclass.asInstanceOf[ParameterizedType].getActualTypeArguments
```

Comment:
@mhamilton723 , without this change, the python wrapper generation for SyntheticDiffInDiffEstimator fails due to multiple level of class hierarchy. This change does not seem to affect other python wrappers, but I didn't fully test it. Hopefully the unit test coverage should be good enough.

Reply:
I tested this change this way: Start a clean branch, only make this one change, and generated the python bindings for the core module, and compared with the generated python bindings without this change from the master branch, and the generated code are equivalent except for version string.

Reply:
Sounds good to me!